src/features/instrument/Instrument.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button } from '@mui/material';
+import './Instrument.css';
+import { Synth, Volume, Sampler, Destination, PolySynth, Transport, Part, Noise, AutoFilter, Player } from 'tone'
+import { BASS_CHORDS, GAMMA_NOTES, MARIO_SAMPLES, MELODY_CHORDS } from '../../constants/notes'
+import { useRef } from 'react';
+const DogSample = require("../../assets/dog-sample.mp3");
+
+const keyStyles = {
+  paddingTop: '4em',
+  paddingBottom: '4em',
+  width: '5em'
+};
+const mediumVolume = new Volume(-15);
+const sampler = new Sampler({
+  E4: DogSample,
+}).chain(mediumVolume, Destination);
+const polySynthSquare = new PolySynth(Synth, {
+  oscillator: {
+    type: "square",
+  },
+}).chain(new Volume(-25), Destination);
+
+export const Instrument = () => {
+  // @TODO: separate to piaoa nd predefined isntruments
+
+  const isTransportStarted = useRef(false);
+  const melodyPart = useRef<Part | null>(null);
+  const bassPart = useRef<Part | null>(null);
+  const noise = useRef<Noise | null>(null);
```

Comment:
dynamic values initialization

its hurts to add .current everywhere

These are the values I am using and changing in several functions. I struggled to decide how that's should be described.

Round 1 - initialized as simple variables with `null values`.
[Link to that commit](https://github.com/laurasilvanavi/test-react-sounds-ts/pull/1/commits/6f6436c54d24fd69dd4ef53cbb776845dfb1db66#diff-dbc1ddb2229691475f99430fb4e92732592bebb1c3fd16f6d7f808098064864cR28-R31 )

Reply:
The main reason you would want to use `useState` over `useRef` would be if you wanted the component to rerender when that value changes.

If you don't need to rerender, useRef is fine. If you want to render changes (like maybe toggling play/pause or mute button states) then state is more appropriate.

I think what you really need is a central place for this song logic. Whether that's a single hook (vs a component) or a sharing the song parts + transport via a context.